test/work_item_repository.go

```diff
@@ -246,6 +253,40 @@ func (fake *WorkItemRepository) ListReturns(result1 []*app.WorkItem, result2 uin
 	}{result1, result2, result3}
 }
 
+func (fake *WorkItemRepository) Fetch(ctx context.Context, c criteria.Expression) (*app.WorkItem, error) {
+	fake.fetchMutex.Lock()
+	fake.fetchArgsForCall = append(fake.fetchArgsForCall, struct {
+		ctx      context.Context
+		criteria criteria.Expression
+	}{ctx, c})
+	fake.recordInvocation("Fetch", []interface{}{ctx, c})
+	fake.fetchMutex.Unlock()
+	if fake.FetchStub != nil {
+		return fake.FetchStub(ctx, c)
+	}
+	return fake.fetchReturns.result1, fake.fetchReturns.result2
+}
+
+func (fake *WorkItemRepository) FetchCallCount() int {
+	fake.fetchMutex.RLock()
+	defer fake.fetchMutex.RUnlock()
+	return len(fake.fetchArgsForCall)
+}
+
+func (fake *WorkItemRepository) FetchArgsForCall(i int) (context.Context, criteria.Expression) {
+	fake.fetchMutex.RLock()
+	defer fake.fetchMutex.RUnlock()
+	return fake.fetchArgsForCall[i].ctx, fake.fetchArgsForCall[i].criteria
+}
+
+func (fake *WorkItemRepository) FetchReturns(result1 *app.WorkItem, result2 error) {
+	fake.FetchStub = nil
+	fake.fetchReturns = struct {
+		result1 *app.WorkItem
+		result2 error
+	}{result1, result2}
+}
+
 func (fake *WorkItemRepository) Invocations() map[string][][]interface{} {
 	fake.invocationsMutex.RLock()
```

Comment:
Actually, this code has been generated by https://github.com/maxbrunsfeld/counterfeiter